cloudfoundry-client/src/main/java/org/cloudfoundry/client/v3/applications/_ListApplicationRoutesRequest.java

```diff
@@ -0,0 +1,58 @@
+package org.cloudfoundry.client.v3.applications;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.cloudfoundry.Nullable;
+import org.cloudfoundry.client.v3.FilterParameter;
+import org.cloudfoundry.client.v3.PaginatedRequest;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class _ListApplicationRoutesRequest extends PaginatedRequest {
+
+    /**
+     * The application id
+     */
+    @JsonIgnore
+    abstract String getApplicationId();
+
+    /**
+     * The hosts
+     */
+    @FilterParameter("hosts")
+    abstract List<String> getHosts();
+
+    /**
+     * The paths
+     */
+    @FilterParameter("paths")
+    abstract List<String> getPaths();
+
+
+    /**
+     * The domain guids
+     */
+    @FilterParameter("domain_guids")
+    abstract List<String> getDomainGuids();
+
+
+    /**
+     * The space guids
+     */
+    @FilterParameter("space_guids")
+    abstract List<String> getSpaceGuids();
+
+    /**
+     * The organization guids
+     */
+    @FilterParameter("organization_guids")
+    abstract List<String> getOrganizationGuids();
+
+    /**
+     * The metadata query
+     */
+    @FilterParameter("label_selector")
+    @Nullable
```

Comment:
FilterParameter has an implicit Nullable so this shouldn't be necessary. Having said that I've noticed I've done it in one or two places, so either we've both made the same mistake or there's a reason for this that I've forgotten. Can you clarify?

Reply:
I have looked how to make the POJOs from the already existing places :) So, if it is mistake, I could remove the @Nullable from everywhere

Reply:
I think remove it for now, if it causes problems it should turn up in integration testing and I'll fix it then. I did just notice that the javadoc you have is wrong for this though.